recipes/default.rb

```diff
@@ -15,3 +15,154 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+# Install PHP + Modules
+node.default['php']['version'] = '7.4'
+node.default['osl-php']['use_ius'] = true
+node.default['osl-php']['use_opcache'] = true
+node.default['osl-php']['php_packages'] =
+  %w(
+    fpm
+    cli
+    imap
+    json
+    mysqlnd
+    mbstring
+  )
+
+include_recipe 'osl-php'
+include_recipe 'osl-apache'
+include_recipe 'osl-apache::mod_php'
+include_recipe 'osl-mysql::client'
+
+# Certificate (wildcard is placeholder for now)
+cert_path = '/etc/pki/tls/certs/wildcard.pem'
+key_path = '/etc/pki/tls/certs/wildcard.key'
+
+# Following this tutorial: https://linuxize.com/post/set-up-an-email-server-with-postfixadmin/
+
+group 'vmail' do
+  gid 5000
+end
+
+user 'vmail' do
+  comment 'Owner of all mailboxes. Used to access emails on this server'
+  home    '/var/mail/vmail'
+  group   'vmail'
+  shell   '/usr/sbin/nologin'
+end
+
+# Download Postfixadmin
+# postfixadmin_download_location = "#{Chef::Config[:file_cache_path]}/postfixadmin.tar.gz"
+
+ark 'postfixadmin' do
+  url postfixadmin_source
+  path '/var/www/postfixadmin'
+  checksum postfixadmin_checksum
+  owner 'apache'
+  strip_components 1
+  action :cherry_pick
+end
+
+# Local Postfix config
+template '/var/www/postfixadmin/config.local.php' do
+  source 'config.local.php.erb'
+  sensitive true
+  variables(
+    db: data_bag_item('mailstore', 'config_mysql')
+  )
+end
+
+# Cache Directory
+directory '/var/www/templates_c' do
+  owner 'apache'
+end
+
+# Postfix
+node.default['osl-postfix']['main'].tap do |main|
+  main['virtual_mailbox_domains'] = 'proxy:mysql:/etc/postfix/sql/mysql_virtual_domains_maps.cf'
+  main['virtual_alias_maps'] = %w(
+    mysql-virtual-alias-maps.cf
+    mysql-virtual-alias-domain-maps.cf
+    mysql-virtual-alias-domain-catchall-maps.cf
+  ).map { |file| 'proxy:mysql:/etc/postfix/sql/' + file }.join(',')
+  main['virtual_mailbox_maps'] = %w(
+    mysql-virtual-mailbox-maps.cf
+    mysql-virtual-alias-domain-mailbox-maps.cf
+  ).map { |file| 'proxy:mysql:/etc/postfix/sql/' + file }.join(',')
+  main['relay_domains'] = 'proxy:mysql:/etc/postfix/sql/mysql-relay-domains.cf'
+  main['transport_maps'] = 'proxy:mysql:/etc/postfix/sql/mysql-transport-maps.cf'
+  main['virtual_mailbox_base'] = '/var/mail/vmail'
+end
+
+include_recipe 'osl-postfix'
+
+directory '/etc/postfix/sql'
+
+# Postfix - MySQL configs
+creds = data_bag_item('sql_creds', 'mysql')
```

Comment:
Shouldn't this be using `data_bag_item('mailstore', 'config_mysql')` like above?